lms/djangoapps/discussion_api/api.py

```diff
@@ -110,6 +110,27 @@ def _is_user_author_or_privileged(cc_content, context):
     )
 
 
+def _update_vote_count(voted, serializer):
+    """
+    Updates the vote_count in the given serializer
+
+    If the user has already voted and `voted == True` do nothing,
+    else +1 to the vote count.
+    If the user has not voted and `voted == False` do nothing,
+    else -1 to the vote count.
+
+    Arguments:
+
+        voted (Bool): Boolean of the new vote status
+        serializer (ThreadSerializer/CommentSerializer): Serializer to update
+    """
+    if voted and not serializer.data["voted"]:
+        serializer.data["vote_count"] += 1
+    elif not voted and serializer.data["voted"]:
+        serializer.data["vote_count"] -= 1
+    return serializer
```

Comment:
No need to return the updated serializer here since the same passed-in serializer is modified.